File: workspace/2023-01-24/fizz-buzz.c/fizz-buzz.c
#include <stdio.h>

void fizz_buzz(int n);
void fizz_buzz_value(int n);

int main(void)
{
    fizz_buzz(100);

    return 0;
}

void fizz_buzz(int n)
{
    int i;

    for (i = 1; i <= n; i++)
        fizz_buzz_value(i);
}

void fizz_buzz_value(int n)
{
    if (n % 15 == 0)
        printf("FizzBuzz\n");
    else if (n % 5 == 0)
        printf("Buzz\n");
    else if (n % 3 == 0)
        printf("Fizz\n");
    else
        printf("%d\n", n);
}

File: workspace/2023/2023-03-01/fizz-buzz.ruby/fizz_buzz.rb

def main
    puts fizz_buzz(100)
end

def fizz_buzz(n)
    (1..n).map do |i|
        fizz_buzz_value(i)
    end
end

def fizz_buzz_value(n)
    case
        when n % 15 == 0
            :FizzBuzz
        when n % 5 == 0
            :Buzz
        when n % 3 == 0
            :Fizz
        else
            n
    end
end

main

